docs/release.md

```diff
@@ -0,0 +1,34 @@
+# Release
+This document describes how to release the ClusterAPI BYOH infrastructure provider.
+
+
+### Current BYOH release: `v0.3.0`
```

Comment:
Will be good to include a table to specify older releases?

Reply:
Do we need to specify releases at all? Then it will be a task every release to update this list.

Reply:
> Do we need to specify releases at all?

+1, adding release here is redundant and might not add much value. Will remove this.